<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen" />
    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen" />
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print" />

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>Simexon by sak042</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h1>Simexon</h1>
        <h2>Simexon: A targeted resequencing (exome sequencing) simulator</h2>
        <a href="https://github.com/sak042/Simexon" class="button"><small>View project on</small>GitHub</a>
      </div>
    </header>

    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h3>Introduction</h3>

<p><strong>Simexon</strong> is a simulator for a targeted resequencing as generally known as exome sequencing. Simexon basically generates a set of <em>artificial</em> DNA fragments for next generation sequencing (NGS) read simulation. In the targeted resequencing, we constraint the genomic regions that are used to generated DNA fragments to be only a part of the entire genome; they are usually exons and/or a few introns and untranslated regions (UTRs).</p>

<h3>Install Simexon</h3>

<p>To run Simexon, Python 2.7 or later is required. To install Python, go to <a href="http://python.org/">http://python.org/</a></p>

<p>Download Simexon using the links in this page, or go to <a href="https://github.com/sak042/Simexon">https://github.com/sak042/Simexon</a></p>

<h3>Requirements</h3>

<p>The following programs are required to run Simexon or to prepare input files:</p>

<ul>
<li>
<strong>pysam</strong> library: go to <a href="http://code.google.com/p/pysam/">http://code.google.com/p/pysam/</a> to install pysam</li>
<li>
<strong>numpy</strong> library: go to <a href="http://numpy.scipy.org/">http://numpy.scipy.org/</a> to install numpy</li>
<li>
<strong>gfServer</strong> and <strong>gfClient</strong>: In probe hybridization mode, Simexon runs more than 100,000 queries again the reference genome. This essentially requires a local blat server. gfServer and gfClient are pre-compiled programs for establishing private blat server on your computer. go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> to download gfServer and gfClient (and set your local path to access the two programs anywhere). For more details about the tools, please refer to <a href="http://genome.ucsc.edu/FAQ/FAQblat.html#blat5">http://genome.ucsc.edu/FAQ/FAQblat.html#blat5</a>
</li>
<li>
<strong>faToTwoBit</strong>: go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> and download faToTwoBit. This is required to convert your FASTA file to .2bit </li>
<li>
<strong>samtools</strong>: samtools is needed to index your sample genome FASTA file (samtools faidx).</li>
<li>
<strong>GemSim</strong> error models: Simexon uses GemSim's empirical error models for NGS read generation. Go to GemSim's project page (<a href="http://sourceforge.net/projects/gemsim/">http://sourceforge.net/projects/gemsim/</a>) to download GemSim. You will find several model files (e.g. ill100v4_p.gzip) under 'models' directory. Save them and remember their location.</li>
</ul><h3>Preparing Input Files</h3>

<p>Simexon requires two major inputs. One is the sample genome sequence, and the other is the target region information.</p>

<ul>
<li>
<strong>Sample genome sequence</strong>: This is a FASTA file (e.g. ref.fa). You will need to index the file and generate .2bit
<pre><code>
&gt;samtools faidx ref.fa
&gt;faToTwoBit ref.fa ref.2bit
</code></pre>
</li>
<li>
<strong>Target region information</strong>: Target regions can be specified by two different ways. 

<ol>
<li>
<strong>Ideal targets</strong>: In ideal target mode, you will provide a list of genomic coordinates in a BED  file (e.g. chr1   798833 799125). Ideal targets of major exome capture platforms are freely available from vendor's website. For Agilent's SureSelect platforms, go to <a href="https://earray.chem.agilent.com/suredesign/">https://earray.chem.agilent.com/suredesign/</a> , For NimbleGen's SeqCap go to <a href="http://www.nimblegen.com/products/seqcap/index.html">http://www.nimblegen.com/products/seqcap/index.html</a> and find BED files under Design and Annotation Files. </li>
<li>
<strong>Probe sequences</strong>: Probe sequences are available for SureSelect platforms in the SureDesign homepage (<a href="https://earray.chem.agilent.com/suredesign/">https://earray.chem.agilent.com/suredesign/</a>). You must register at their site. After loggin in, go to Find Designs and select Agilent Catalog at the menu tab. You will be able to download all information of currently available platforms including ideal target BED files and probe sequence text files.<br>
</li>
</ol>
</li>
</ul><h3>Running Simexon</h3>

<p>The basic synopsis of Simexon command is like below:</p>

<pre><code>
&gt;python Simexon1.py -R ref.fa -B target.bed -n 1000000 -l 100 -M model.gzip -z -o result
&gt;python Simexon2.py -R ref.fa -P probe.txt.fa -B probe_match.txt.fa.psl -n 1000000 -l 76 -M model.gzip -pz -o result2
</code></pre>

<h3>Authors and Contributors</h3>

<p>You can <a href="https://github.com/blog/821" class="user-mention">@mention</a> a GitHub username to generate a link to their profile. The resulting <code>&lt;a&gt;</code> element will link to the contributor's GitHub Profile. For example: In 2007, Chris Wanstrath (<a href="https://github.com/defunkt" class="user-mention">@defunkt</a>), PJ Hyett (<a href="https://github.com/pjhyett" class="user-mention">@pjhyett</a>), and Tom Preston-Werner (<a href="https://github.com/mojombo" class="user-mention">@mojombo</a>) founded GitHub.</p>

<h3>Support or Contact</h3>

<p>Having trouble with Pages? Check out the documentation at <a href="http://help.github.com/pages">http://help.github.com/pages</a> or contact <a href="mailto:support@github.com">support@github.com</a> and weâ€™ll help you sort it out.</p>
        </section>

        <aside id="sidebar">
          <a href="https://github.com/sak042/Simexon/zipball/master" class="button">
            <small>Download</small>
            .zip file
          </a>
          <a href="https://github.com/sak042/Simexon/tarball/master" class="button">
            <small>Download</small>
            .tar.gz file
          </a>

          <p class="repo-owner"><a href="https://github.com/sak042/Simexon"></a> is maintained by <a href="https://github.com/sak042">sak042</a>.</p>

          <p>This page was generated by <a href="pages.github.com">GitHub Pages</a> using the Architect theme by <a href="http://twitter.com/jasonlong">Jason Long</a>.</p>
        </aside>
      </div>
    </div>

  
  </body>
</html>