<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Wessim by sak042</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 class="header">Wessim</h1>
        <p class="header">Wessim: Whole Exome Sequencing SIMulator using in silico exome capture</p>

        <ul>
          <li class="download"><a class="buttons" href="https://github.com/sak042/Wessim/zipball/master">Download ZIP</a></li>
          <li class="download"><a class="buttons" href="https://github.com/sak042/Wessim/tarball/master">Download TAR</a></li>
          <li><a class="buttons github" href="https://github.com/sak042/Wessim">View On GitHub</a></li>
        </ul>

        <p class="header">This project is maintained by <a class="header name" href="https://github.com/sak042">sak042</a></p>


      </header>
      <section>
        <h3>Introduction</h3>

<p><strong>Wessim</strong> is a simulator for a targeted resequencing as generally known as exome sequencing. Wessim basically generates a set of <em>artificial</em> DNA fragments for next generation sequencing (NGS) read simulation. In the targeted resequencing, we constraint the genomic regions that are used to generated DNA fragments to be only a part of the entire genome; they are usually exons and/or a few introns and untranslated regions (UTRs).</p>

<h3>Install Wessim</h3>

<p>Download Wessim using the links in this page, or go to <a href="https://github.com/sak042/Wessim">https://github.com/sak042/Wessim</a><br>
To run Wessim, Python 2.7 or later is required. To install Python, go to <a href="http://python.org/">http://python.org/</a></p>

<h3>Requirements</h3>

<p>The following programs are required to run Wessim or to prepare input files:</p>

<ul>
<li>
<strong>pysam</strong> library: go to <a href="http://code.google.com/p/pysam/">http://code.google.com/p/pysam/</a> to install pysam</li>
<li>
<strong>numpy</strong> library: go to <a href="http://numpy.scipy.org/">http://numpy.scipy.org/</a> to install numpy</li>
<li>
<strong>gfServer</strong> and <strong>gfClient</strong>: In probe hybridization mode, Wessim runs more than 100,000 queries against the reference genome. This essentially requires a local blat server. gfServer and gfClient are pre-compiled programs for establishing private blat server on your computer. go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> to download gfServer and gfClient (and set your local path to access the two programs anywhere). For more details about the tools, please refer to <a href="http://genome.ucsc.edu/FAQ/FAQblat.html#blat5">http://genome.ucsc.edu/FAQ/FAQblat.html#blat5</a>
</li>
<li>
<strong>faToTwoBit</strong>: go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> and download faToTwoBit. This is required to convert your FASTA file to .2bit </li>
<li>
<strong>samtools</strong>: samtools is needed to index your sample genome FASTA file (samtools faidx).</li>
<li>
<strong>GemSim</strong> error models: Wessim uses GemSim's empirical error models for NGS read generation. Go to GemSim's project page (<a href="http://sourceforge.net/projects/gemsim/">http://sourceforge.net/projects/gemsim/</a>) to download GemSim. You will find several model files (e.g. ill100v4_p.gzip) under 'models' directory. Save them and remember their location.</li>
</ul><h3>Preparing Input Files</h3>

<p>Wessim requires two major inputs. One is the sample genome sequence, and the other is the target region information.</p>

<ul>
<li>
<strong>Sample genome sequence</strong>: This is a FASTA file (e.g. ref.fa). You will need to index the file and generate .2bit
<pre><code>
&gt;samtools faidx ref.fa
&gt;faToTwoBit ref.fa ref.2bit
</code></pre>
</li>
<li>
<strong>Target region information</strong>: Target regions can be specified by two different ways.

<ol>
<li>
<strong>Ideal targets</strong>: In ideal target mode, you will provide a list of genomic coordinates in a BED  file (e.g. chr1   798833 799125). Ideal targets of major exome capture platforms are freely available from vendor's website. For Agilent's SureSelect platforms, go to <a href="https://earray.chem.agilent.com/suredesign/">https://earray.chem.agilent.com/suredesign/</a> . You must register at their site. After logging in, go to Find Designs and select Agilent Catalog at the menu tab. You will be able to download all information of currently available platforms including ideal target BED files and probe sequence text files.   For NimbleGen's SeqCap go to <a href="http://www.nimblegen.com/products/seqcap/index.html">http://www.nimblegen.com/products/seqcap/index.html</a> and find BED files under Design and Annotation Files. </li>
<li>
<strong>Probe sequences</strong>: Probe sequences are available for SureSelect platforms in the SureDesign homepage (<a href="https://earray.chem.agilent.com/suredesign/">https://earray.chem.agilent.com/suredesign/</a>) (see above). Usually those files are named "[platform]_probe.txt"</li>
</ol>
</li>
</ul><h3>Running Wessim</h3>

<p>There are two main scripts in the package - Wessim1.py and Wessim2.py. You will use Wessim1 if you are using a BED file for target regions (ideal target approach). However,  it is highly recommended to use Wessim2 (probe hybridization approach) when the probe sequence is available; it is much more realistic and recovers the statistics of real data. Two other scripts that start with 'Prep' are used to preparing Wessim2 inputs. You can ignore remaining scripts that start with '__sub'; main Wessim programs will execute these sub scripts automatically.</p>

<p>The basic synopsis of Wessim1 is like below:</p>

<pre><code>
# Run Wessim1 in ideal target mode
&gt;python Wessim1.py -R ref.fa -B target.bed -n 1000000 -l 100 -M model.gzip -z -o result -t 4
</code></pre> 

<p>This will generate <em>result.fastq.gz</em> (single-end mode / gzip compressed) using 4 threads (CPU cores).</p>

<p>For Wessim2:</p>

<pre><code>
# Generate a FASTA file of probe sequence
&gt;python Prep_Probe2Fa.py probe.txt (this generates probe.txt.fa)
# Establish your local blat server
&gt;gfServer start -canStop localhost 6666 ref.2bit
# Run blat search to generate the match list
&gt;python Prep_BlatSearch.py -R ref.2bit -P probe.txt.fa -o probe.txt.fa.psl
# Run Wessim2 in probe hybridization mode.
&gt;python Wessim2.py -R ref.fa -P probe.txt.fa -B probe_match.txt.fa.psl -n 1000000 -l 76 -M model.gzip -pz -o result
</code></pre>

<p>This will generate <em>result_1.fastq.gz</em> and <em>result_2.fastq.gz</em> (paired-end mode / gzip compressed).</p>

<h3>Running in metagenomic mode</h3>

<p>You can use more than one genome as your template. To run Wessim in metagenomic mode, you can just write a simple description file that ends with (.meta). Use the meta description file at the place of your reference FASTA file (-R option)</p>

<pre><code>
&gt;python Wessim2.py -R reference.meta -P probe.txt.fa -B probe_match.txt.fa.psl -n 1000000 -l 76 -M model.gzip -pz -o result
</code></pre>

<p>The format of a tab-delimited .meta file is like below,</p>

<pre><code>
genome1.fasta  abundance_of_genome1
genome2.fasta  abundance_of_genome2
...
genomeN.fasta  abundance_of_genomeN
</code></pre>

<p>For example, you can use,</p>

<pre><code>
/data/reference/ref1.fa    0.2
/data/reference/ref2.fa    0.4
/data/reference/ref3.fa    0.4
</code></pre>

<p>Please make sure the overall abundances add up to 1</p>

<h3>Wessim Options</h3>

<p>You can use '-h' for detailed help in command line.</p>

<pre><code>Mandatory input files (for Wessim1 and Wessim2 in common):
  -R FILE     faidx-indexed (R)eference genome FASTA file
For Wessim1 only:
  -B FILE     Target region .(B)ED file
For Wessim2 only:
  -P FILE     (P)robe sequence FASTA file
  -B FILE     (B)lat matched probe regions .PSL file

Parameters for exome capture:
  -f INT      mean (f)ragment size. this corresponds to insert size when sequencing in paired-end mode. [200]
  -d INT      standard (d)eviation of fragment size [50]
  -m INT      (m)inimum fragment length [read_length + 20]
  -x INT      slack margin of the given boundaries [0] (only for Wessim1)
  -w INT      penalty (w)eight for indel in the hybridization [2] (only for Wessim2)

Parameters for sequencing:
  -p          generate paired-end reads [single]
  -n INT      total (n)umber of reads
  -l INT      read (l)ength (bp)
  -M FILE     GemSim (M)odel file (.gzip)
  -t INT      number of (t)hreaded subprocesses [1]

Output options:
  -o FILE     (o)utput file header. ".fastq.gz" or ".fastq" will be attached automatically. Output will be splitted into two files in paired-end mode
  -z          compress output with g(z)ip [false]
  -q INT      (q)uality score offset [33]
  -v          (v)erbose; print out intermediate messages.
</code></pre>

<h3>Support or Contact</h3>

<p>For GitHub use, check out the documentation at <a href="http://help.github.com/pages">http://help.github.com/pages</a> or contact <a href="mailto:support@github.com">support@github.com</a> and weâ€™ll help you sort it out.</p>
      </section>
      <footer>
        <p><small>Hosted on <a href="https://pages.github.com">GitHub Pages</a> using the Dinky theme</small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
		          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("rachmani");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>