<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Simexon by sak042</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 class="header">Simexon</h1>
        <p class="header">Simexon: A targeted resequencing (exome sequencing) simulator</p>

        <ul>
          <li class="download"><a class="buttons" href="https://github.com/sak042/Simexon/zipball/master">Download ZIP</a></li>
          <li class="download"><a class="buttons" href="https://github.com/sak042/Simexon/tarball/master">Download TAR</a></li>
          <li><a class="buttons github" href="https://github.com/sak042/Simexon">View On GitHub</a></li>
        </ul>

        <p class="header">This project is maintained by <a class="header name" href="https://github.com/sak042">sak042</a></p>


      </header>
      <section>
        <h3>Introduction</h3>

<p><strong>Simexon</strong> is a simulator for a targeted resequencing as generally known as exome sequencing. Simexon basically generates a set of <em>artificial</em> DNA fragments for next generation sequencing (NGS) read simulation. In the targeted resequencing, we constraint the genomic regions that are used to generated DNA fragments to be only a part of the entire genome; they are usually exons and/or a few introns and untranslated regions (UTRs).</p>

<h3>Install Simexon</h3>

<p>To run Simexon, Python 2.7 or later is required. To install Python, go to <a href="http://python.org/">http://python.org/</a></p>

<p>Download Simexon using the links in this page, or go to <a href="https://github.com/sak042/Simexon">https://github.com/sak042/Simexon</a></p>

<h3>Requirements</h3>

<p>The following programs are required to run Simexon or to prepare input files:</p>

<ul>
<li>
<strong>pysam</strong> library: go to <a href="http://code.google.com/p/pysam/">http://code.google.com/p/pysam/</a> to install pysam</li>
<li>
<strong>numpy</strong> library: go to <a href="http://numpy.scipy.org/">http://numpy.scipy.org/</a> to install numpy</li>
<li>
<strong>gfServer</strong> and <strong>gfClient</strong>: In probe hybridization mode, Simexon runs more than 100,000 queries again the reference genome. This essentially requires a local blat server. gfServer and gfClient are pre-compiled programs for establishing private blat server on your computer. go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> to download gfServer and gfClient (and set your local path to access the two programs anywhere). For more details about the tools, please refer to <a href="http://genome.ucsc.edu/FAQ/FAQblat.html#blat5">http://genome.ucsc.edu/FAQ/FAQblat.html#blat5</a>
</li>
<li>
<strong>faToTwoBit</strong>: go to <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</a> and download faToTwoBit. This is required to convert your FASTA file to .2bit </li>
<li>
<strong>samtools</strong>: samtools is needed to index your sample genome FASTA file (samtools faidx).</li>
<li>
<strong>GemSim</strong> error models: Simexon uses GemSim's empirical error models for NGS read generation. Go to GemSim's project page (<a href="http://sourceforge.net/projects/gemsim/">http://sourceforge.net/projects/gemsim/</a>) to download GemSim. You will find several model files (e.g. ill100v4_p.gzip) under 'models' directory. Save them and remember their location.</li>
</ul><h3>Preparing Input Files</h3>

<p>Simexon requires two major inputs. One is the sample genome sequence, and the other is the target region information.</p>

<ul>
<li>
<p>Sample genome sequence: This is a FASTA file (e.g. ref.fa). You will need to index the file and generate .2bit
</p>
<pre>
samtools faidx ref.fa
faToTwoBit ref.fa ref.2bit</pre>
</li>
<li>
<p>Target region information: Target regions can be specified by two different ways. </p>

<ul>
<li>Ideal target mode: In ideal target mode, you will provide a list of genomic coordinates in a BED file (e.g. chr1   798833 799125). Ideal targets of major exome capture platforms are freely available from vendor's website. For Agilent's SureSelect platforms, go to <a href="https://earray.chem.agilent.com/suredesign/">https://earray.chem.agilent.com/suredesign/</a> , For NimbleGen's SeqCap go to <a href="http://www.nimblegen.com/products/seqcap/index.html">http://www.nimblegen.com/products/seqcap/index.html</a> and find BED files under Design and Annotation Files. </li>
</ul>
</li>
</ul><h3>Authors and Contributors</h3>

<p>You can <a href="https://github.com/blog/821" class="user-mention">@mention</a> a GitHub username to generate a link to their profile. The resulting <code>&lt;a&gt;</code> element will link to the contributor's GitHub Profile. For example: In 2007, Chris Wanstrath (<a href="https://github.com/defunkt" class="user-mention">@defunkt</a>), PJ Hyett (<a href="https://github.com/pjhyett" class="user-mention">@pjhyett</a>), and Tom Preston-Werner (<a href="https://github.com/mojombo" class="user-mention">@mojombo</a>) founded GitHub.</p>

<h3>Support or Contact</h3>

<p>Having trouble with Pages? Check out the documentation at <a href="http://help.github.com/pages">http://help.github.com/pages</a> or contact <a href="mailto:support@github.com">support@github.com</a> and weâ€™ll help you sort it out.</p>
      </section>
      <footer>
        <p><small>Hosted on <a href="https://pages.github.com">GitHub Pages</a> using the Dinky theme</small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
		
  </body>
</html>